react-router/react-router.d.ts

```diff
@@ -247,7 +247,7 @@ declare namespace ReactRouter {
       isActive: (pathOrLoc: H.LocationDescriptor, indexOnly?: boolean) => boolean
     }
 
-    function withRouter<C extends React.ComponentClass<any>>(component: C): C
+    function withRouter<C extends React.ComponentClass<any>>(component: C): C & InjectedRouter
```

Comment:
I agree with you! My definition was incorrect. I like your proposed change. I have updated the PR.